### PR TITLE
Pytorch frontend op cross_entropy

### PR DESCRIPTION
**Pull request by notoraptor**
Hi @breuleux @abergeron @ethancaballero 

This is a PR to implement pytorch frontend operation `cross_entropy`.

@ethancaballero I did not yet try to implement all cross entropy function parameters (weight, size_average, ignore_index, reduce and reduction), I just implemented input and target parameters. Could you check if I modified right files in right places ? Should I implement all parameters ?

Thanks!

### CODE REVIEW

File: myia/frontends/pytorch.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 from collections import OrderedDict
 
 import torch
+import torch.nn.functional
```
Comment (ethancaballero) [inline myia/frontends/pytorch.py]: you don't need to import torch.nn.functional

File: myia/frontends/pytorch_functions.py [hunk 2]
```diff
@@ -19,6 +19,7 @@
 from functools import reduce
 
 import torch
+import torch.nn.functional
```
Comment (ethancaballero) [inline myia/frontends/pytorch_functions.py]: you don't need to import torch.nn.functional

File: tests/frontends/test_pytorch_ops.py [hunk 3]
```diff
@@ -311,6 +311,17 @@ def test_torch_conv2d__group3(inp, w, b):
     return torch.sum(value)
 
 
+@mt(
+    fwd_and_bwd(nn.Parameter(torch.Tensor(torch.randn(3, 5))),
+                nn.Parameter(torch.randint(5, (3,)),
+                             requires_grad=False)),
+    run(torch.randn(3, 5), torch.randint(5, (3,))),
+    backend=backend_no_relay
+)
```
Comment (ethancaballero) [inline tests/frontends/test_pytorch_ops.py]: This should be:
```
@mt(
    fwd_and_bwd(nn.Parameter(torch.Tensor(torch.randn(3, 5))),
                nn.Parameter(torch.randint(5, (3,)),
                             requires_grad=False)),
)
```

File: myia/frontends/pytorch_functions.py [hunk 4]
```diff
@@ -268,6 +269,14 @@ def conv2d(input, weight, bias=None, stride=1, padding=0, dilation=1,
     return ret
 
 
+@core
+def cross_entropy(input, target):
+    """Map of method torch.nn.functional.cross_entropy."""
+    a = torch.nn.functional.log_softmax(input, 1)
+    b = torch.nn.functional.nll_loss(a, target)
+    return b
```
Comment (ethancaballero) [inline myia/frontends/pytorch_functions.py]: This should be:
```
@core
def cross_entropy(input, target, reduction='mean'):
    """Map of method torch.nn.functional.cross_entropy."""
    a = torch.nn.functional.log_softmax(input, 1)
    b = torch.nn.functional.nll_loss(a, target, reduction=reduction)
    return b
```

**Comment by ethancaballero** [comment]
The additional parameter that needs to be added to cross_entopy is `reduction`. I've added how to so in the line comment of mine.

File: myia/frontends/pytorch.py (hunk 1 above)
Comment (notoraptor) [reply myia/frontends/pytorch.py]: Done!

File: myia/frontends/pytorch_functions.py (hunk 2 above)
Comment (notoraptor) [reply myia/frontends/pytorch_functions.py]: Done!

File: tests/frontends/test_pytorch_ops.py (hunk 3 above)
Comment (notoraptor) [reply tests/frontends/test_pytorch_ops.py]: Done! I used `fwd_and_bwd_no_relay` instead of just `fwd_and_bwd`, because I got some compilation errors with relay, and I guess we don't need relay to work here as we are in Pytorch frontend, do we?

I also duplicated the test for each currently supported reduction (none, mean and sum)

File: myia/frontends/pytorch_functions.py (hunk 4 above)
Comment (notoraptor) [reply myia/frontends/pytorch_functions.py]: Done!
